### ISSUE TITLE
No error raised with a specific syntax

### ISSUE THREAD

**Issue by FredSeloger** [SonarSource/sonar-csharp#848]
### Description

We were facing to a production issue due to object construction describes below. Sonar doesn’t raise any error if a property  is called with a null object. 

### Repro steps

1. Use an object construction as below :
new myObj { A1= B.a1, A2= D.a2, A3= X.a3 });

2. Execute Sonar on this code.

### Expected behavior

Sonar raises an error with a description : verify if the object B is null before call property a1

### Actual behavior

No error is raised because there is not test on the object B in order to verify if the object is null

### Related information

* SonarC#   Version : 6.5 - plugin SonarC# : 6.3
* Visual   Studio Version : Professionnal 2015

**Comment by michalb-sonar** [comment]
Thanks for the report @FredSeloger.

SonarC# is currently limited to the scope of the method, so this code would not raise:
```
void Foo(B obj) { obj.ToString(); }
Foo(null);
```

The analyzer must know the value was null at some point to report the problem. For example, the value was assigned/initialized to null or there was an check for null, hinting the value can be null. If that did not happen, we make no assumptions, as the rule would be extremely noisy if we raised on every property/field access. For example, for public methods - raising issues for every single parameter that is not validated to be not null would create large amount of noise.

I imagine the problem you are seeing is due to the current limitation of analyzing only the method body. If you think that assumption is not correct, please share more code - you might have possibly found a bug.
